quipucords/api/report/view.py

```diff
@@ -9,193 +9,193 @@
-            if report is not None:
-                return Response(report)
-            return Response(status=status.HTTP_404_NOT_FOUND)
-
-    @staticmethod
-    def validate_filters(filters):
-        """Check the combination of filters are allowed.
-
-        :param filters: report filters to checks
-        :raises: Raises validation error if the combination of
-        filters is not allowed.
-        """
-        filter_count = len(filters.keys())
-        fc_found = filters.get('fact_collection_id', None)
-        group_count_found = filters.get('group_count', None)
-        if group_count_found is not None:
-            if fc_found is not None and filter_count > 2:
-                error = {
-                    'query_params': [_(messages.REPORT_GROUP_COUNT_FILTER)]
-                }
-                raise ValidationError(error)
-            elif fc_found is None and filter_count > 1:
-                error = {
-                    'query_params': [_(messages.REPORT_GROUP_COUNT_FILTER)]
-                }
-                raise ValidationError(error)
-
-    @staticmethod
-    def build_grouped_report(fact_collection_id, fingerprints, group):
-        """Create a count report based on the fingerprints and the group.
-
-        :param fact_collection_id: the identifer for the fact collection
-        :param fingerprints: The system fingerprints used in the group count
-        :param group: the field to group and count on
-        :returns: json report data
-        :raises: Raises validation error group_count on non-existent field.
-        """
-        try:
-            # Group by field and count
-            counts_by_group = fingerprints.values(
-                group).annotate(total=Count(group))
-        except FieldError:
-            msg = _(messages.REPORT_GROUP_COUNT_FIELD % (group))
+logger = logging.getLogger(__name__)
+authentication_enabled = os.getenv('QPC_DISABLE_AUTHENTICATION') != 'True'
+
+if authentication_enabled:
+    auth_classes = (ExpiringTokenAuthentication,
+                    SessionAuthentication)
+    perm_classes = (IsAuthenticated,)
+else:
+    auth_classes = ()
+    perm_classes = ()
+
+
+# pylint: disable=C0103, W0613, R0201
+@api_view(['GET'])
+@authentication_classes(auth_classes)
+@permission_classes(perm_classes)
+@renderer_classes((JSONRenderer, BrowsableAPIRenderer,
+                   FactCollectionCSVRenderer))
+def details(request, pk=None):
+    """Lookup and return a details system report."""
+    if pk is not None:
+        if not is_int(pk):
+            error = {
+                'report_id': [_(messages.COMMON_ID_INV)]
+            }
+            raise ValidationError(error)
+    detail_data = get_object_or_404(FactCollection.objects.all(), pk=pk)
+    serializer = FactCollectionSerializer(detail_data)
+    json_details = serializer.data
+    http_accept = request.META.get('HTTP_ACCEPT')
+    if http_accept and 'text/csv' not in http_accept:
+        json_details.pop('csv_content')
```

Comment:
You should do json_details.pop('csv_content', None)

Reply:
done